src/psyclone/tests/psyir/backend/c_test.py

```diff
@@ -306,15 +309,21 @@ def test_cw_unaryoperator():
                  (UnaryOperation.Operator.REAL, '(float)a'))
 
     for operator, expected in test_list:
+        # pylint: disable=protected-access
         unary_operation._operator = operator
+        # pylint: enable=protected-access
         assert cwriter(unary_operation) in expected
 
     # Test that an unsupported operator raises an error
     # pylint: disable=abstract-method, too-few-public-methods
     class Unsupported():
         '''Dummy class'''
-    # pylint: enable=abstract-method, too-few-public-methods
+        def __init__(self):
+            pass
+    # pylint: enable=too-few-public-methods
+    # pylint: disable=protected-access
```

Comment:
Difficult question that, it had me puzzled for a while. Problem: I was sometimes running pylint directly from a subdirectory of tests, not tests itself. Then it would not pick up the .pylintrc file in tests, and give me those warnings again. But when we turned tests into a module (by adding some __init__.py files), then pylint will search the directory tree and find tests/.pylintrc .

So I've added some __init__.py files in tests subdirectories to make sure wherever we run pylint from, it will find tests/.pylintrc. I then also removed all unnecessary pylint directives in all tests.

You've added a .pylintrc file to nemo, which contained just "../.pylintrc". This seems to be invalid, both python2 and 3 pylints would just crash (when running in the nemo subdir). I've removed that file and also added __init__.py so that it will pick up tests/.pylintrc.

Reply:
Ah, that makes sense.